utils/network.go

```diff
@@ -62,123 +140,70 @@ func (n *Network) MarshalJSON() ([]byte, error) {
-	case "GOERLI", "goerli":
-		*n = Goerli
-	case "GOERLI2", "goerli2":
-		*n = Goerli2
-	case "INTEGRATION", "integration":
-		*n = Integration
-	case "SEPOLIA", "sepolia":
-		*n = Sepolia
-	case "SEPOLIA_INTEGRATION", "sepolia-integration":
-		*n = SepoliaIntegration
-	default:
-		return ErrUnknownNetwork
+	predefinedNetworks := map[string]Network{
+		"MAINNET":             Mainnet,
+		"mainnet":             Mainnet,
+		"GOERLI":              Goerli,
+		"goerli":              Goerli,
+		"GOERLI2":             Goerli2,
+		"goerli2":             Goerli2,
+		"INTEGRATION":         Integration,
+		"integration":         Integration,
+		"SEPOLIA":             Sepolia,
+		"sepolia":             Sepolia,
+		"SEPOLIA-INTEGRATION": SepoliaIntegration,
+		"sepolia-integration": SepoliaIntegration,
 	}
-	return nil
-}
-
-func (n *Network) Type() string {
-	return "Network"
-}
 
-func (n *Network) UnmarshalText(text []byte) error {
-	return n.Set(string(text))
+	if network, ok := predefinedNetworks[s]; ok {
+		*n = network
+		return nil
+	}
+	return ErrUnknownNetwork
 }
 
-// baseURL returns the base URL without endpoint
-func (n Network) baseURL() string {
-	switch n {
-	case Goerli:
-		return "https://alpha4.starknet.io/"
-	case Mainnet:
-		return "https://alpha-mainnet.starknet.io/"
-	case Goerli2:
-		return "https://alpha4-2.starknet.io/"
-	case Integration:
-		return "https://external.integration.starknet.io/"
-	case Sepolia:
-		return "https://alpha-sepolia.starknet.io/"
-	case SepoliaIntegration:
-		return "https://integration-sepolia.starknet.io/"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
+func (n *Network) SetCustomNetwork(s string) error {
+	*n = Network{}
+	if err := n.UnmarshalJSON([]byte(s)); err != nil {
+		return fmt.Errorf("%w: %s", ErrInvalidCustomNetworkJSONStr, err)
 	}
+	return n.Validate()
 }
 
-// FeederURL returns URL for read commands
-func (n Network) FeederURL() string {
-	return n.baseURL() + "feeder_gateway/"
-}
+func (n *Network) Validate() error {
+	if !(n.Name == "CUSTOM" || n.Name == "custom") {
+		return ErrUnknownNetwork
+	}
 
-// GatewayURL returns URL for write commands
-func (n Network) GatewayURL() string {
-	return n.baseURL() + "gateway/"
+	validate := validator.Validator()
+	return validate.Struct(n)
 }
 
-func (n Network) ChainIDString() string {
-	switch n {
-	case Goerli, Integration:
-		return "SN_GOERLI"
-	case Mainnet:
-		return "SN_MAIN"
-	case Goerli2:
-		return "SN_GOERLI2"
-	case Sepolia:
-		return "SN_SEPOLIA"
-	case SepoliaIntegration:
-		return "SN_INTEGRATION_SEPOLIA"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
+func (n *Network) UnmarshalJSON(data []byte) error {
+	type Alias Network
+	aux := &struct {
+		CoreContractAddress string `json:"core_contract_address"`
+		*Alias
+	}{
+		Alias: (*Alias)(n),
 	}
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+	n.CoreContractAddress = common.HexToAddress(aux.CoreContractAddress)
+	return nil
```

Comment:
Why do we need to override `UnmarshalJSON` to get a new format? Apologies if this was discussed on one of the prior PRs, but my intuition says the default behavior should be good here.

Reply:
Its because strings aren't automatically unmarshalled into common.Address types (at least when I tried it last - but I can look into it again)

Reply:
Update: this is no longer needed, so has been removed